scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/license/LicenseSchedulingPolicy.java

```diff
@@ -290,25 +280,31 @@ public boolean isTaskExecutable(NodeSet selectedNodes, EligibleTaskDescriptor ta
             final HashSet<String> requiredLicensesSet = new HashSet<String>(Arrays.asList(requiredLicenses.split(",")));
             for (String requiredLicense : requiredLicensesSet) {
 
-                if (!containsJob(eligibleJobsDescriptorsLicenses.get(requiredLicense),
-                                 ((EligibleTaskDescriptorImpl) task).getInternal().getJobId()) &&
+                if (!licenseSynchronization.containsJobId(requiredLicense,
+                                                          ((EligibleTaskDescriptorImpl) task).getInternal()
+                                                                                             .getJobId()
+                                                                                             .value()) &&
                     !canGetToken(requiredLicense)) {
                     logger.debug("Token for " + requiredLicense + " not available, keep task pending");
                     return false;
                 }
             }
-            logger.debug("Can be executed since all tokens are available for licenses " + requiredLicenses);
+            logger.debug("Task can be executed since all tokens are available for licenses " + requiredLicenses);
             for (String requiredLicense : requiredLicensesSet) {
-                if (!containsJob(eligibleJobsDescriptorsLicenses.get(requiredLicense),
-                                 ((EligibleTaskDescriptorImpl) task).getInternal().getJobId())) {
-
-                    eligibleTasksDescriptorsLicenses.get(requiredLicense).add(task);
-                    remainingTokens.put(requiredLicense, remainingTokens.get(requiredLicense) - 1);
+                if (!licenseSynchronization.containsJobId(requiredLicense,
+                                                          ((EligibleTaskDescriptorImpl) task).getInternal()
+                                                                                             .getJobId()
+                                                                                             .value())) {
+
+                    licenseSynchronization.addTaskToLicense(requiredLicense, task.getTaskId().toString());
+                    licenseSynchronization.acquireToken(requiredLicense);
```

Comment:
it looks like addTaskToLicense and acquireToken could be done in a single call (the second being called from the first)

Reply:
Ok.